File: cpp/leetcode/twosum.cpp
#include <iostream>
#include <map>
#include <vector>

using namespace std;

class Solution {
 public:
  vector<int> twoSum(vector<int> &nums, int target) {
    map<int, int> m{};
    vector<int> result{};
    for (int i = 0; i < nums.size(); i++) {
      if (m.find(target - nums[i]) != m.end()) {
        result.push_back(m.find(target - nums[i])->second);
        result.push_back(i);
        break;
      }
      m.insert(pair<int, int>(nums[i], i));
    }
    return result;
  }
};

File: rust/variables/src/two_sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut d = HashMap::new();

        for (i, &num) in nums.iter().enumerate() {
            let v = target - num;
            if let Some(t_i) = d.get(&v) {
                return vec![*t_i as i32, i as i32];
            }
            d.insert(num, i);
        }
        vec![]
    }
}
